src/main/java/com/example/auctionapp/security/JwtUtil.java

```diff
@@ -52,4 +52,9 @@ public static Boolean validateToken(String token, UserDetails userDetails, Strin
         final String username = extractUsername(token, secretKey);
         return (username.equals(userDetails.getUsername()) && !isTokenExpired(token, secretKey));
     }
+
+    public static Boolean validateTokenWithoutExpiration(String token, UserDetails userDetails, String secretKey) {
+        final String username = extractUsername(token, secretKey);
+        return (username.equals(userDetails.getUsername()));
+    }
```

Comment:
Refreshing token by user name is not secure enough. Anyone could save an old token and refresh it with this logic.
One approach would be for the API to return a one use token on login, which can be used to refresh.